code/controllers/subsystems/processing/statuseffects.dm

```diff
@@ -0,0 +1,188 @@
+						break
+					continue
+				effect.doEffect()
+
+// Adds a status effect , if theres already a effect but the new one is gonna last longer the original one gets extended
+proc/addStatusEffect(mob/target, effectType, duration)
+	if(QDELETED(target))
+		return FALSE
+	// Not bad per say , but we shouldn't be using this subsystem for this scenario
+	if(duration < 1)
+		CRASH("Attempt to create statusEffect with a duration below 1 : [duration]")
+	if(!GLOB.globalEffects[effectType])
+		CRASH("Invalid effectType received : [effectType]")
+	var/ef_type = GLOB.globalEffects[effectType]
+	var/datum/statusEffect/createdEffect = new ef_type()
+	if(!(createdEffect.flags & SE_FLAG_UNIQUE ))
+		if(length(SSstatusEffects.affectedMobs[ref(target)]))
+			var/datum/statusEffect/existingEffect = locate(ef_type) in SSstatusEffects.affectedMobs[ref(target)]
+			if(existingEffect)
+				if(existingEffect.duration + existingEffect.startingTime < world.time + duration)
+					existingEffect.startingTime = world.time
+					existingEffect.duration = duration
+				return TRUE
+
+	createdEffect.startingTime = world.time
+	createdEffect.duration = duration
+	createdEffect.mobReference = ref(target)
+	if(length(SSstatusEffects.affectedMobs[ref(target)]))
+		SSstatusEffects.affectedMobs[ref(target)] += createdEffect
+	else
+		SSstatusEffects.affectedMobs[ref(target)] += list(createdEffect)
+	createdEffect.onStart()
+	return TRUE
+
+proc/removeStatusEffect(mob/target, effectType)
+	if(QDELETED(target))
+		return 0
+	if(!effectType)
+		return 0
+	for(var/datum/statusEffect/effect as anything in SSstatusEffects.affectedMobs[ref(target)])
+		if(effect.identifier == effectType)
+			// guarantees removal on next tick
+			effect.duration = -1
+
+proc/hasStatusEffect(mob/target, effectType)
+	if(QDELETED(target))
+		return FALSE
+	if(!effectType)
+		return FALSE
+	for(var/datum/statusEffect/effect as anything in SSstatusEffects.affectedMobs[ref(target)])
+		if(effect.identifier == effectType)
+			return TRUE
+	return FALSE
+
+proc/getStatusEffectDuration(mob/target, effectType)
+	. = 0
+	if(QDELETED(target))
+		return 0
+	if(!effectType)
+		return 0
+	for(var/datum/statusEffect/effect as anything in SSstatusEffects.affectedMobs[ref(target)])
+		if(effect.identifier == effectType)
+			if(effect.flags & SE_FLAG_UNIQUE)
+				if(. < effect.duration + effect.startingTime - world.time)
+					. = effect.duration + effect.startingTime - world.time
+			else
+				return effect.duration + effect.startingTime - world.time
+	return .
+
+// expect a list if its a unique effect
+proc/getStatusEffect(mob/target, effectType)
+	var/list/return_list = list()
+	if(QDELETED(target))
+		return FALSE
+	if(!effectType)
+		return FALSE
+	for(var/datum/statusEffect/effect as anything in SSstatusEffects.affectedMobs[ref(target)])
+		if(effect.identifier == effectType)
+			if(!(effect.flags & SE_FLAG_UNIQUE))
+				return effect
+			return_list.Add(effect)
+	return return_list
+
+
+/datum/statusEffect
```

Comment:
Does this need to be derived from `/datum`? If all you need are the vars below, it would function fine as its own `/status_effect` object.